### ISSUE TITLE
Regression: the new Block Inserter doesn't use headings to identify block sections any longer

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#22859]
**Describe the bug**
The new Block Inserter doesn't use headings to identify block sections any longer.

In the old inserter, each expandable panel was also a `h2` heading.

<img width="735" alt="Screenshot 2020-06-03 at 15 59 24" src="https://user-images.githubusercontent.com/1682452/83648429-52ccb600-a5b6-11ea-8bf5-65cab16ff821.png">

This greatly helped screen reader users to jump through the various blocks groups because headings are one of the main navigation mechanisms for screen readers.

In the new block inserter, the expandable panels are gone but what really impacts accessibility is that the blocks groups "titles" aren't headings any longer.

<img width="449" alt="Screenshot 2020-06-03 at 15 58 24" src="https://user-images.githubusercontent.com/1682452/83648698-a5a66d80-a5b6-11ea-8d2c-31d32a291252.png">

These titles are just plain text with no semantics. Thus, screen reader users aren't able to jump through sections any longer. 

I'm not sure what is the reasoning behind the headings removal, I'd appreciate anyone pointing me to the relevant discussion. On my side, I'd just like to note that it's 2020 and I'd appreciate if anyone would to their best to avoid to throw unsemantic text into the UI. 

The new implementation uses just `div` and `span` elements for these text, which is, I'm sorry to say it, just not acceptable in a project that aims to be accessible.

```
<div class="block-editor-inserter__panel-header">
	<span class="block-editor-inserter__panel-title">Most used</span>
</div>

```

**To reproduce**
- activate the Gutenberg plugini 8.2.1
- open the main inserter
- inspect the source in your browser's dev tools
- observe the blocks groups titles aren't headings any longer

**Expected behavior**
The various blocks groups to be identified with a `h2` heading.

**Editor version (please complete the following information):**
- WordPress version: 5.4
- Gutenberg plugin version 8.2.1

Note:
I consider this an accessibility regression and I'd like to kindly ask to not release the new block inserter in this state in WordPress 5.5. The headings should be restored first.

**Comment by afercia** [comment]
Same applies to the new "Patterns" tab, where all the section titles are just unsemantic text. Headings would be very beneficial:

TEXT
HERO
COLUMNS
BUTTONS
GALLERY
FEATURES
TESTIMONIALS
UNCATEGORIZED

**Comment by gaambo** [comment]
I'd be happy to take a try at this, since I also just added a PR for #22863 and those seem connected.
